src/utils.ts

```diff
@@ -113,17 +113,22 @@ const fetchWithHeaders = async (
 ): Promise<Response> => {
   const defaultHeaders = {
     'Content-Type': 'application/json',
-    Accept: 'application/json',
+    Accept: 'application/json', 
     'User-Agent': `ollama-js/${version} (${getPlatform()})`,
-  }
+  } as HeadersInit
 
   if (!options.headers) {
     options.headers = {}
   }
 
+  // Filter out default headers from custom headers
+  const customHeaders = Object.fromEntries(
```

Comment:
Bringing to parity with python sdk impl. For ref: https://github.com/ollama/ollama-python/pull/313